### DISCUSSION TITLE
zh/guide/advanced/search

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#92] [Announcements]
# zh/guide/advanced/search

如何使用 按照以下步骤开启搜索: 1. 转到索引页，选择一个搜索索引，并单击保存; 2. 保存索引后，单击构建索引来构建索引; 3. 现在你可以通过点击页面右上角的搜索块或使用快捷键Ctrl + K来搜索文件。 不同搜索索引之间的差异 database:按数据库搜索，它使用现有的 data.db。它将创建一个新表，记录父目录、名称和每个对象的大小，但搜...

https://alist.nn.ci/zh/guide/advanced/search.html

**Comment by Reduction77** [comment]
Nice Job, I update the first time because of this!

**Reply by Gaobang100** [reply]
Haha, me too!

**Comment by Gaobang100** [comment]
太棒了，由于我在分类上似乎有着天生的劣势，我常常找不到想要的文件。这个功能很好解决了我的问题！（而且动画很赞❤）
原来这个功能在最新版才有……我找了半天“索引”页，还当成了“index.html”（我：？？Alist有静态网页？）能在前面提示一下更新吗qwq
哦，还有个小请求：由于我存储空间里的目录太长，导致我复制或移动文件的时候常常要点开一大长串的文件树……能不能将复制移动里的文件树“展开当前工作目录”，而不是只有一个“root”（还需要继续点开）
或者是我更期待能不能有一个“拖动模式”，就是像在文件资源管理器那样可以拖动以“复制”（按住Ctrl），“移动”（按住Shift），或创建符号链接（按住“Alt”），这样一定会大大提高工作效率（文件资源管理器的Webdav挂载老是“未响应”）。
最后，谢谢开发者啦❤❤❤^-^

**Reply by yhn666** [reply]
才知道alist更新了，~~我说为啥我找不到索引：(~~，尤其是我把我所有网盘全绑上了，找东西成了困难事~~哭~~

**Comment by ssdswi** [comment]
数据库好像并不会自动更新

**Reply by FantasyFish** [reply]
我自动更新和手动更新都不行

**Comment by Richman881** [comment]
我本地内网 搭建的alist，文件也是本地磁盘的，对象计数:288419，用数据库索引，搜索特别慢,单次搜索都要2分钟左右。请问一下，ALIST的性能对硬件环境要求很高么，或是有什么样的推荐。

**Comment by FantasyFish** [comment]
不知道是不是我索引深度数值我设置的太大了（我设置的300），构建了一夜找到1000多万个文件，然后我的服务器的磁盘就满了（占了我20G的空间），按说不应该占用那么多空间才对，有人知道为啥么

**Reply by ltxhhz** [reply]
可能是有循环链接的目录

**Comment by FantasyFish** [comment]
为啥无法更新索引呢？自动更新和手动更新都不行

**Comment by 21AND11** [comment]
请问怎样将MySQL作为数据库呀！请大神指教！^^

**Comment by Kvinam** [comment]
支持postgresql吗

**Comment by daqyt** [comment]
Index is running怎么解决啊！更新配置都用不了

**Comment by dfjmmm** [comment]
如果能支持正则表达式的搜索就好了

**Comment by asdf1319964f** [comment]
删除的数据能在索引中自动删除吗 还是需要在此重构

**Comment by qinguoluo** [comment]
在开了自动更新索引的情况下，把文件从一个文件夹移动到另外一个文件夹，你会发现目标文件夹会更新索引，源文件夹索引似乎没有动，因为你在完成文件移动后再去搜索这个文件，源文件夹和目标文件夹下面都有这个文件。这算不算一个bug？我的alist版本是v3.27.0。

**Comment by wongz** [comment]
忽略路径好像不支持相对路径

**Comment by shaohuazi** [comment]
您好，V3搜索似乎无法正常工作。
搜索出来的文件直接点击，路径会带一个%3Ffrom=search，而文件路径会把这个东西加进去，导致object not found。

**Comment by smalleagleLuo** [comment]
这个索引功能，有个严重的问题：A目录以及所有子目录都设置了需要密码访问，开启索引后，能够搜索到A目录下的子目录以及文件，并且能够不需要密码直接访问！

**Comment by linzeliang1222** [comment]
我使用mysql作为数据库，然后都设置页面展示顺序都乱了，这是为什么呢
![](https://oss.linzeliang.com/file/2024/03/17/20_53_44_RKmGNu.png)

**Comment by wxyShine** [comment]
不是管理员 登录搜不到内容，管理员登录正常搜

**Comment by KB64ba** [comment]
小雅有什么办法可以共享元信息进行共享索引，这样就不用重新扫一遍避免风控

**Comment by cdesign89** [comment]
请问如何只索引指定文件夹呢？（除了用忽略路径把其他全忽略）
我的情况是网盘上直接挂载下来了，但我只需要为视频文件夹增加索引功能，其他文件太多了，也不一定用得上。
想通过只索引特定文件夹，减轻负担。

不知道是有这个功能我没找到，还是只有忽略路径这个功能？
如果是只能忽略，是否能考虑做成类似于【黑白名单】的模式？
感谢开发者！！♥

**Comment by luyabing** [comment]
mysql的连接，能不能直接放在设置里，nas里的配置文件找不到，太麻烦了。
